core/ui/src/main/java/com/google/samples/apps/nowinandroid/core/ui/NewsResourceCardList.kt

```diff
@@ -47,8 +47,8 @@ fun <T> LazyListScope.newsResourceCardItems(
     key = { newsResourceMapper(it).id },
     itemContent = { item ->
         val newsResource = newsResourceMapper(item)
-        val launchResourceIntent =
-            Intent(Intent.ACTION_VIEW, Uri.parse(newsResource.url))
+        val resourceUrl = Uri.parse(newsResource.url)
+        val backgroundColor = MaterialTheme.colorScheme.background.toArgb()
```

Comment:
Just wondering if there's a way we could move the getting of `backgroundColor` and `context` to inside `launchCustomChromeTab` to avoid having to write this code each time we want to call `launchCustomChromeTab`.

Reply:
That's a really good question! I thought about it, and I couldn't really find a suitable solution for it because both `context` and `backgroundColor` are coming from composable functions and we can't really get the `context` out of nowhere, and the same situation for `backgroundColor`. Even if we find a solution for the `context`, then it would probably be a dangerous one because of leaking context and so on. Please also notice that `launchCustomChromeTab()` is not a composable function and maybe it can't even be one!

Anyways, the only thing that came to my mind was to write an extension function to avoid the repetition of the above method;

`fun Context.launchCustomChromeTab()`

Please let me know what you think! Thanks

Reply:
i couldn't find a way to call `MaterialTheme.colorScheme` without making `launchCustomChromeTab` a Composable as well. 

I think generally, chrome custom tabs doesn't work cleanly with Compose code. Setting a customized `close` button is also difficult with Compose. Perhaps something to file for the chrome custom tab team.